plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java

```diff
@@ -214,10 +214,12 @@ public InputStream getInputStream(String account, String container, String blob)
                 // uri.getPath is of the form /container/keyPath.* and we want to strip off the /container/
                 // this requires 1 + container.length() + 1, with each 1 corresponding to one of the /
                 final String blobPath = uri.getPath().substring(1 + container.length() + 1);
-                final BlobProperties properties = ((CloudBlockBlob) blobItem).getProperties();
-                final String name = blobPath.substring(keyPath.length());
-                logger.trace(() -> new ParameterizedMessage("blob url [{}], name [{}], size [{}]", uri, name, properties.getLength()));
-                blobsBuilder.put(name, new PlainBlobMetaData(name, properties.getLength()));
+                if (blobItem instanceof CloudBlockBlob) {
```

Comment:
should we use the more generic `CloudBlob` here? Do you know under which conditions this is returning `CloudBlobDirectory`?

Reply:
> should we use the more generic CloudBlob here?

Can't do it, we need to get the size off of the object and the generic blob doesn't have that on its interface.

> Do you know under which conditions this is returning CloudBlobDirectory?

Sure, simple:

Whenever we do a list by prefix and it catches a directory we error out. E.g. if someone accidentally created a directory in a shard dir (I know it's a little far fetched for someone to manually created a dir somewhere but: 1. still we shouldn't completely shut down if they did. 2. Maybe it's not that far fetched given that people could be messing around with the nested dir setting of their repos).

Reply:
> Can't do it, we need to get the size off of the object and the generic blob doesn't have that on its interface.

The `getProperties()` method is on `CloudBlob`, not `CloudBlockBlob`.

Note that we should also make the `listBlobsByPrefix` method consistent (and have tests for it) across all implementations. AFAICS, for HDFS it also returns directories...

Perhaps better to keep this in #42653 and add the tests there.

Reply:
Ah sorry my bad, I mixed up the kinds of blobs :D Sure let's keep it in #42653 then (the already has tests for this scenario :) that's how I found this :P). I'll weaken the cast there then + make the tests a little more explicit for this case -> closing here :)